test/Microsoft.AspNetCore.SignalR.Tests/HubConnectionHandlerTests.cs

```diff
@@ -2019,6 +2019,73 @@ public int Serialize(ref byte[] bytes, int offset, T value, IFormatterResolver f
             }
         }
 
+        [Fact]
+        public async Task ConnectionNotTimedOutIfClientNeverPings()
+        {
+            var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider(services =>
+                services.Configure<HubOptions>(options =>
+                   options.ClientTimeoutInterval = TimeSpan.FromMilliseconds(100)));
+            var connectionHandler = serviceProvider.GetService<HubConnectionHandler<MethodHub>>();
+
+            using (var client = new TestClient(new JsonHubProtocol()))
+            {
+                var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
+                await client.Connected.OrTimeout();
+                // This is a fake client -- it doesn't auto-ping to signal
+
+                await Task.Delay(120);
```

Comment:
I'm guessing it is important that the delay be 120 to be over the ClientTimeoutInterval? Could you add some more comments to the tests explaining what is being simulated.